### DISCUSSION TITLE
How do I build a private release .zip? (snaphot)

### DISCUSSION THREAD

**Discussion by vicos59** [geodesicsolutions-community/geocore-community#153] [Q&A] [answered]
@jonyo Would you tell me how I can build a zip file. That would allow me to get the latest main branch installed on my test domain so I can begin converting my templates in order to complete my migration/upgrade from 7.5.1.  Thanks!

**Comment by jonyo** [comment, marked as answer]
If you are running it in docker then you have the files already built in src/

But it may have extras like the config file and user images uploaded when testing...

Anyways, to build, in a command line in the docket container, run:

```
composer build-release
```

If I remembered the right name, that should do it.  Worst case if I remembered wrong it will just say unknown command or something.

**Reply by jonyo** [reply]
If that did work it will build the zip files in a build/ folder.

**Reply by jonyo** [reply]
@vicos59 if that worked for you, since this is posted in the "q & a" section, you can mark it as the answer...  I think it may end up the first question with a marked answer in the discussions!

**Reply by vicos59** [reply]
OK, I want to wait until the 2 open PRs get merged. Then, I will try it and mark this thread accordingly.

**Reply by vicos59** [reply]
It works as described. However, please see my note below about which CLI you need to use on Windows; or maybe rather, which CLI you should not use.

**Reply by jonyo** [reply]
Yup, maybe I should have made that part stand out more:

> ... in the docket container ...

**Comment by vicos59** [comment]
Note: On windows, `composer build-release` will not work from within the Visual Studio Code Terminal (at least as configured out of the box on my pc):

```
PS C:\Users\vicos59\Github Projects\geodesicsolutions-community\geocore-community> composer build-release
composer : The term 'composer' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify 
that the path is correct and try again.
At line:1 char:1
+ composer build-release
+ ~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (composer:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

```

**It did run from Docker CLI. Go to Docker Desktop, click on the CLI icon ( '>_' ) to the right of 'geocore-community-web-1'**

4/four zip files are created in the build/ folder at the top level of the repo.

![build-dir](https://user-images.githubusercontent.com/6634041/157695598-af4d365b-78c4-428a-b299-89c8b4401572.JPG)

**Reply by jonyo** [reply]
Oh you answered your own thing, yes you need to do it inside the docker container if you don't have composer set up on the host.

**Comment by vicos59** [comment]
>Oh you answered your own thing

Yeah, I do that a lot 🤣
